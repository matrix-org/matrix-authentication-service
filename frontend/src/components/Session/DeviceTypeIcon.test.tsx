// Copyright 2023 The Matrix.org Foundation C.I.C.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

// @vitest-environment happy-dom

import { composeStory } from "@storybook/react";
import { render, cleanup } from "@testing-library/react";
import { describe, it, expect, afterEach } from "vitest";

import Meta, { Unknown, Desktop, Mobile, Web } from "./DeviceTypeIcon.stories";

describe("<DeviceTypeIcon />", () => {
  afterEach(cleanup);

  it("renders unknown device type", () => {
    const Component = composeStory(Unknown, Meta);
    const { container } = render(<Component />);
    expect(container).toMatchSnapshot();
  });
  it("renders mobile device type", () => {
    const Component = composeStory(Mobile, Meta);
    const { container } = render(<Component />);
    expect(container).toMatchSnapshot();
  });
  it("renders desktop device type", () => {
    const Component = composeStory(Desktop, Meta);
    const { container } = render(<Component />);
    expect(container).toMatchSnapshot();
  });
  it("renders Web device type", () => {
    const Component = composeStory(Web, Meta);
    const { container } = render(<Component />);
    expect(container).toMatchSnapshot();
  });
});
