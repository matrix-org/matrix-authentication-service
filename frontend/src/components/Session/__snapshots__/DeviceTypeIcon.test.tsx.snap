// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<DeviceTypeIcon /> > renders mobile device type 1`] = `
<div>
  <svg
    aria-label="Mobile"
    class="_deviceTypeIcon_e677aa"
    fill="currentColor"
    height="1em"
    viewBox="0 0 24 24"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M7 23c-.55 0-1.02-.196-1.412-.587A1.926 1.926 0 0 1 5 21V3c0-.55.196-1.02.588-1.413A1.926 1.926 0 0 1 7 1h10c.55 0 1.02.196 1.413.587.39.393.587.863.587 1.413v18c0 .55-.196 1.02-.587 1.413A1.926 1.926 0 0 1 17 23H7Zm0-5h10V6H7v12Z"
    />
  </svg>
</div>
`;

exports[`<DeviceTypeIcon /> > renders pc device type 1`] = `
<div>
  <svg
    aria-label="Computer"
    class="_deviceTypeIcon_e677aa"
    fill="currentColor"
    height="1em"
    viewBox="0 0 24 24"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M4 18c-.55 0-1.02-.196-1.413-.587A1.926 1.926 0 0 1 2 16V5c0-.55.196-1.02.587-1.413A1.926 1.926 0 0 1 4 3h16c.55 0 1.02.196 1.413.587.39.393.587.863.587 1.413v11c0 .55-.196 1.02-.587 1.413A1.926 1.926 0 0 1 20 18H4Zm0-2h16V5H4v11Zm-2 5a.967.967 0 0 1-.712-.288A.968.968 0 0 1 1 20c0-.283.096-.52.288-.712A.967.967 0 0 1 2 19h20c.283 0 .52.096.712.288.192.191.288.429.288.712s-.096.52-.288.712A.968.968 0 0 1 22 21H2Z"
    />
  </svg>
</div>
`;

exports[`<DeviceTypeIcon /> > renders tablet device type 1`] = `
<div>
  <svg
    aria-label="Tablet"
    class="_deviceTypeIcon_e677aa"
    fill="currentColor"
    height="1em"
    viewBox="0 0 24 24"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M4 20c-.55 0-1.02-.196-1.413-.587A1.926 1.926 0 0 1 2 18V6c0-.55.196-1.02.587-1.412A1.926 1.926 0 0 1 4 4h16c.55 0 1.02.196 1.413.588.391.391.587.862.587 1.412v12c0 .55-.196 1.02-.587 1.413A1.926 1.926 0 0 1 20 20H4Zm0-2h16V8H4v10Z"
    />
  </svg>
</div>
`;

exports[`<DeviceTypeIcon /> > renders unknown device type 1`] = `
<div>
  <svg
    aria-label="Unknown device type"
    class="_deviceTypeIcon_e677aa"
    fill="currentColor"
    height="1em"
    viewBox="0 0 24 24"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M5 21c-.55 0-1.02-.196-1.413-.587A1.926 1.926 0 0 1 3 19V5c0-.55.196-1.02.587-1.413A1.926 1.926 0 0 1 5 3h14c.55 0 1.02.196 1.413.587.39.393.587.863.587 1.413v14c0 .55-.196 1.02-.587 1.413A1.926 1.926 0 0 1 19 21H5Zm0-2h14V5H5v14Z"
    />
    <path
      d="M11 10a1 1 0 1 1 1.479.878c-.31.17-.659.413-.94.741-.286.334-.539.8-.539 1.381a1 1 0 0 0 2 .006.3.3 0 0 1 .057-.085 1.39 1.39 0 0 1 .382-.288A3 3 0 1 0 9 10a1 1 0 1 0 2 0Zm1.999 3.011v-.004.005ZM12 17a1 1 0 1 0 0-2 1 1 0 0 0 0 2Z"
    />
  </svg>
</div>
`;
